template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/raz/RangerRazBaseConfigProvider.java

```diff
@@ -40,13 +47,20 @@ public List<String> getRoleTypes() {
 
     @Override
     protected List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
+        List<ApiClusterTemplateConfig> roleConfigs = new ArrayList<>();
         String cdhVersion = source.getBlueprintView().getProcessor().getVersion().orElse("");
         CloudPlatform cloudPlatform = source.getCloudPlatform();
-        if (!Strings.isNullOrEmpty(cdhVersion) && isRazTokenConfigurationSupported(cdhVersion, cloudPlatform, source.getStackType())) {
+        if (!Strings.isNullOrEmpty(cdhVersion) && isRazConfigurationForServiceTypeSupported(cdhVersion, cloudPlatform, source.getStackType())) {
             String safetyValveValue = getSafetyValveProperty(RANGER_RAZ_BOOTSTRAP_SERVICETYPES, getServiceType(cloudPlatform));
-            return List.of(config(RANGER_RAZ_SITE_XML_ROLE_SAFETY_VALVE, safetyValveValue));
+            roleConfigs.add(config(RANGER_RAZ_SITE_XML_ROLE_SAFETY_VALVE, safetyValveValue));
         }
-        return List.of();
+        if (isRazConfigurationForRazRoleSupported(source.getProductDetailsView().getCm().getVersion(), cloudPlatform, source.getStackType())) {
```

Comment:
It is clearly seen here as well that this role config will be added only in the case of GCP